src/main/java/io/cryostat/net/web/http/api/v1/RecordingDeleteHandler.java

```diff
@@ -102,34 +94,24 @@ public boolean isAsync() {
     @Override
     public void handleAuthenticated(RoutingContext ctx) throws Exception {
         String recordingName = ctx.pathParam("recordingName");
-        fs.listDirectoryChildren(savedRecordingsPath).stream()
-                .filter(saved -> saved.equals(recordingName))
-                .map(savedRecordingsPath::resolve)
-                .findFirst()
-                .ifPresentOrElse(
-                        path -> {
-                            try {
-                                if (!fs.exists(path)) {
-                                    throw new HttpStatusException(404, recordingName);
-                                }
-                                fs.deleteIfExists(path);
-                                notificationFactory
-                                        .createBuilder()
-                                        .metaCategory(NOTIFICATION_CATEGORY)
-                                        .metaType(HttpMimeType.JSON)
-                                        .message(Map.of("recording", recordingName))
-                                        .build()
-                                        .send();
-                            } catch (IOException e) {
-                                throw new HttpStatusException(500, e.getMessage(), e);
-                            } finally {
-                                reportService.delete(recordingName);
-                            }
-                            ctx.response().setStatusCode(200);
-                            ctx.response().end();
-                        },
-                        () -> {
-                            throw new HttpStatusException(404, recordingName);
-                        });
+        try {
+            recordingArchiveHelper.deleteRecording(recordingName).get();
+            notificationFactory
```

Comment:
Would it make sense to also extract this notification emission into the archive helper's `deleteRecording` method?

Reply:
Sure, I'll do the same for the patch save.